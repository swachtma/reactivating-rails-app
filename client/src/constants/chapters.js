export const LOAD_CHAPTERS = "LOAD_CHAPTERS";