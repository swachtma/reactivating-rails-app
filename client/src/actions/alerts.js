import * as constants from "../constants/alerts";

export const addAlert = (message, level = constants.lvls[0], scope = constants.scopes[0]) => ({
  type: constants.ADD_ALERT, payload: {message, level, scope, timestamp: Date.now()}
});

export const clearAlerts = () => ({ type: constants.CLEAR_ALERTS }) ;