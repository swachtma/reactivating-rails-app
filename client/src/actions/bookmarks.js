import * as constants from "../constants/bookmarks";

export const setBookmarkLocations = (chapter_id) => ({
  type: constants.SET_BOOKMARK_LOCATIONS, payload: {chapter: chapter_id}
});

export const setBookmarkOffered = (bool) => ({
  type: constants.SET_BOOKMARK_OFFERED, payload: {prompt_shown: bool}
});