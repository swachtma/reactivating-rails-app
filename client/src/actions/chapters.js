import * as constants from '../constants/chapters'

export const loadChapters = (payload) => ({ type: constants.LOAD_CHAPTERS, payload })
