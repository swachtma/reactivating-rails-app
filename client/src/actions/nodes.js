import * as constants from '../constants/nodes'

export const loadNodes = (payload) => ({ type: constants.LOAD_NODES, payload })
