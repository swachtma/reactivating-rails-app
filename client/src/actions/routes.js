import * as routes from '../constants/settings';

export const routeHome = () => ({type: routes.HOME_ROUTE});

export const routeChapter = (chapter_id) => (
  {type: routes.CHAPTER_ROUTE, payload: { chapter_id }}
);