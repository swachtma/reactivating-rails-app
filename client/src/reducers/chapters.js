import * as constants from '../constants/chapters';

let default_state = [];

export const sortChapters = (chapters) => {
  return chapters.sort((a,b) =>{
    return a.id - b.id;
  });
};

const chaptersReducer = function(state = default_state,action){
  switch (action.type) {
    case(constants.LOAD_CHAPTERS):
      return [...sortChapters(action.payload)];
    default:
      return state;
  }
};

export default chaptersReducer;