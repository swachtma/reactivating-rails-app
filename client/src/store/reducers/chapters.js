import * as constants from '../../constants/chapters';

let default_state = [];

export const sortChapters = (chapters) => 
  chapters.sort((a,b) => a.id - b.id);

export default function chaptersReducer(state = default_state,action){
  switch (action.type) {
    case(constants.LOAD_CHAPTERS):
      return [...sortChapters(action.payload)];
    default:
      return state;
  }
}