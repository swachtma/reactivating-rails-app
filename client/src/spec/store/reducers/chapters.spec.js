/* global expect */
import  chaptersReducer, { sortChapters } from '../../../store/reducers/chapters';
import * as actions from '../../../actions/chapters';

let defaultState = [];
let newState = [{title: "Test chapter"}];
let sampleChapters = [
  {id: 7, title: "Second"},
  {id: 8, title: "Third"},
  {id: 6, title: "First"}
];

describe("chapters Reducer", ()=>{
  test("sortChapters orders by id", ()=>{
    expect(sortChapters(sampleChapters)).toMatchSnapshot();
  });
  
  test("LOAD_CHAPTERS overwrites chapters array",()=>{
    expect(chaptersReducer(defaultState,actions.loadChapters(newState))).toEqual(newState);
  });
});