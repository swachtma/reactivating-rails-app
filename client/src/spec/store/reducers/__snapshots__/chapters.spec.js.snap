// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chapters Reducer sortChapters orders by id 1`] = `
Array [
  Object {
    "id": 6,
    "title": "First",
  },
  Object {
    "id": 7,
    "title": "Second",
  },
  Object {
    "id": 8,
    "title": "Third",
  },
]
`;
